crates/python/src/qpu/result_data.rs

```diff
@@ -43,4 +43,27 @@ impl PyQpuResultData {
     fn readout_values(&self, py: Python<'_>) -> PyResult<HashMap<String, PyReadoutValues>> {
         self.as_inner().readout_values().to_python(py)
     }
+
+    fn asdict(&self, py: Python<'_>) -> PyResult<Py<PyDict>> {
+        let dict = PyDict::new(py);
+        dict.set_item("mappings", self.mappings(py)?)?;
+        dict.set_item(
+            "readout_values",
+            self.as_inner()
+                .readout_values()
+                .iter()
+                .map(|(register, values)| {
+                    (
+                        register.to_string(),
+                        match values {
+                            ReadoutValues::Integer(values) => values.to_object(py),
+                            ReadoutValues::Real(values) => values.to_object(py),
+                            ReadoutValues::Complex(values) => values.to_object(py),
+                        },
```

Comment:
Is there a more concise way of doing this? Only thing I could think of is some kind of trait on `ReadoutValues`, but then we'd bleed python functionality into the Rust crate.

Reply:
This is honestly pretty concise, imo.

Reply:
Do the Python values lose the numeric-type information?

Reply:
...oh, I see, `values` has a different `Item` type depending on the numeric type it represents.